pex/pex_builder.py

```diff
@@ -123,13 +123,13 @@ def __entry_point_from_filename__(filename):
     __entry_point__ = __loader__.archive
   else:
     try:
-      from pkgutil import ImpLoader
-      if isinstance(__loader__, ImpLoader):
-        __entry_point__ = __entry_point_from_filename__(__loader__.get_filename())
-    except ImportError:
       from importlib.abc import Loader
       if isinstance(__loader__, Loader):
         __entry_point__ = __entry_point_from_filename__(__loader__.get_filename())
+    except ImportError:
```

Comment:
This order is a bit better since it stresses the code now as well as getting rid of warnings under newer Pythons. It penalizes existing Python 2.7 users with a small speed bump for the cold cache case in the ~ms range though.

Reply:
Alright - took the feedback but went further given the prompt to think.